docs/usage/configuration/user_authentication/refresh_tokens.md

```diff
@@ -0,0 +1,124 @@
+      is widespread.
+      * Users with clients that support refresh tokens will still benefit from the
+        added security; it's not possible to downgrade a session to using long-lived
+        access tokens so this effectively gives users the choice.
+    * In a closed environment where all users use known clients, this may not be
+      an issue as the homeserver administrator can know if the clients have refresh
+      token support.
+
+
+## Configuration Guide
+
+The following configuration options, in the `registration` section, are related:
+
+* `session_lifetime`: maximum length of a session, even if it's refreshed.
+  In other words, the client must log in again after this time period.
+  In most cases, this can be unset (infinite) or set to a long time (years or months).
+* `refreshable_access_token_lifetime`: lifetime of access tokens that are created
+  by clients supporting refresh tokens.
+  This should be short; a good value might be 5 minutes (`5m`).
+* `nonrefreshable_access_token_lifetime`: lifetime of access tokens that are created
+  by clients which don't support refresh tokens.
+  Make this short if you want to force use of refresh tokens.
+  Make this long if you don't want to inconvenience users of clients which don't
+  support refresh tokens.
+* `refresh_token_lifetime`: lifetime of refresh tokens.
+  Unless you want to log inactive sessions out, it is often fine to use a long
+  value here or even leave it unset (infinite).
+  Beware that making it too short will inconvenience clients that do not connect
+  very often, including mobile clients and clients of infrequent users.
+
+
+### Using refresh token expiry to log out inactive sessions
+
+If you'd like to force sessions to be logged out upon inactivity, you can enable
+refreshable access token expiry and refresh token expiry.
+
+Assuming that clients keep their access token valid whilst they are active, and
+that you set `refresh_token_lifetime` to some value *R* and
+`refreshable_access_token_lifetime` to some value *A* (which is not bigger than *R*),
+then:
+ 1. sessions may become logged out if they are inactive for more than *R - A*.
+    (Explanation: if a client goes offline just before it was about to refresh its
+    access token, then it must come online before the refresh token expires.)
+ 2. sessions will always become logged out if they are inactive for more than *R*.
+    (Explanation: if a client refreshes its access token just before going offline,
+    then returns just before the refresh token expires, it will be able to continue
+    its session.)
+
+Clients are able to refresh more frequently than strictly necessary in order to
+avoid the case described by point (1) above.
+
+As an example, if refresh tokens are configured to expire after 20 minutes and
+access tokens are configured to expire after 5 minutes, then sessions will definitely
+end if inactive for 20 minutes, but may end if inactive for as few as 15 minutes.
```

Comment:
Something like

Suppose I am granted an access token and refresh token at 12:00. These last for 5 and 20 minutes respectively.
   - I can keep using my access token until 12:05.
   - If I go inactive at 12:00 immediately, I must use my refresh token before it expires at 12:20. This gives me a 20 minute window.
   - If I go inactive at 12:02, I must use my refresh token before it expires at 12:20. This gives me an 18 minute window.
   - If I go inactive at 12:05, I must use my refresh token before it expires at 12:20. This gives me a 15 minute window.

Feels like an information overload. I wonder if it'd be simpler to just say something like 

> Each time they login or refresh a token, clients must refresh again within time`R` to avoid being logged out.

Hmm. I'm not really convinced by my effort there.

Reply:
I'm attempting (maybe not very well) to paint a picture of how homeserver admins can configure their servers to have which user-visible behaviours.
What clients should do should hopefully be obvious from the spec.

Basically, if a homeserver admin (or one of our customers...) has a problem statement like 'I want sessions to get logged out automatically if inactive for 15 minutes', this document should help them decide what values they need and hopefully explain why you get a bit of a *range* of times (with the widest range assuming that clients are lazy and only refresh at the last possible moment)...